doc/source/whatsnew/v0.24.0.txt

```diff
@@ -194,6 +194,7 @@ Other Enhancements
 - :meth:`Index.to_frame` now supports overriding column name(s) (:issue:`22580`).
 - New attribute :attr:`__git_version__` will return git commit sha of current build (:issue:`21295`).
 - Compatibility with Matplotlib 3.0 (:issue:`22790`).
+- A newly constructed empty :class:`DataFrame` of integers will now only be cast to ``float64`` if an index is specified (:issue:`22858`)
```

Comment:
Hmm I think the wording `DataFrame of integers` is somewhat misleading as there aren't actually any integers here; the type is just passed explicitly during construction and not subsequently coerced.